### PR TITLE
Use stdpath() to determine rplugin manifest path

### PR DESCRIPTION
**Pull request by jamessan**
I don't actually use any rplugins, so it would be good for someone that does to
double-check this.

### CODE REVIEW

File: runtime/plugin/rplugin.vim [hunk 1]
```diff
@@ -11,14 +11,8 @@ function! s:GetManifestPath() abort
...
 
-  let dest = has('win32') ? '$LOCALAPPDATA' : '$XDG_DATA_HOME'
-  if !exists(dest)
-    let dest = has('win32') ? '~/AppData/Local' : '~/.local/share'
-  endif
-
-  let dest = fnamemodify(expand(dest), ':p')
+  let dest = stdpath('data')
   if !empty(dest)
-    let dest .= ('/' ==# dest[-1:] ? '' : '/') . 'nvim'
```
Comment (justinmk) [inline runtime/plugin/rplugin.vim]: Eek. We weren't using nvimdata on Windows, so this could be a speedbump.

**Review by mhinz — APPROVED** [review]

File: runtime/plugin/rplugin.vim (hunk 1 above)
Comment (jamessan) [reply runtime/plugin/rplugin.vim]: We already have a function to migrate old data.  Could do the same here for the non-`nvim-data` paths on Windows.

Comment (justinmk) [reply runtime/plugin/rplugin.vim]: SGTM

File: runtime/plugin/rplugin.vim [hunk 2]
```diff
@@ -29,22 +23,32 @@ function! s:GetManifestPath() abort
...
   if !filereadable(manifest)
     " Check if an old manifest file exists and move it to the new location.
-    let old_manifest = s:GetOldManifestPath()
-    if filereadable(old_manifest)
-      call rename(old_manifest, manifest)
-    endif
+    for old_manifest in s:GetOldManifestPaths()
+      if filereadable(old_manifest)
+        call rename(old_manifest, manifest)
+        break
```
Comment (justinmk) [inline runtime/plugin/rplugin.vim]: maybe it should not break?

File: runtime/plugin/rplugin.vim [hunk 3]
```diff
@@ -29,22 +23,32 @@ function! s:GetManifestPath() abort
...
 
 function! s:GetManifest() abort
   let manifest = s:GetManifestPath()
   if !filereadable(manifest)
     " Check if an old manifest file exists and move it to the new location.
-    let old_manifest = s:GetOldManifestPath()
-    if filereadable(old_manifest)
-      call rename(old_manifest, manifest)
-    endif
+    for old_manifest in s:GetOldManifestPaths()
```
Comment (justinmk) [inline runtime/plugin/rplugin.vim]: Currently this tries:

```
C:\Users\foo\AppData\Local/nvim
C:\Users\foo\AppData\Local\nvim/.init.vim-rplugin~
```

but I think we want the first one to be:

```
C:\Users\foo\AppData\Local/nvim/rplugin.vim
```

Comment (justinmk) [reply runtime/plugin/rplugin.vim]: This fixes it:

      return [substitute(pseudostdpath, '[/\\]\=$', '/', '') . 'nvim/rplugin.vim', origpath]

Comment (jamessan) [reply runtime/plugin/rplugin.vim]: Fixed.

**Pull request merged**